modin/pandas/test/test_groupby.py

```diff
@@ -1216,11 +1216,16 @@ def eval_shift(modin_groupby, pandas_groupby):
         pandas_groupby,
         lambda groupby: groupby.shift(periods=-3),
     )
-    eval_general(
-        modin_groupby,
-        pandas_groupby,
-        lambda groupby: groupby.shift(axis=1, fill_value=777),
-    )
+    # Disabled for `BaseOnPython` because of the issue with `getitem_array`.
+    # groupby.shift internally masks the source frame with a Series boolean mask,
+    # doing so ends up in the `getitem_array` method, that is broken for `BaseOnPython`:
+    # https://github.com/modin-project/modin/issues/3701
```

Comment:
Boolean masking inside `shift` is done [here](https://github.com/modin-project/modin/blob/ece98a6122c604b03f58ff34f4cb1fd3b2293915/modin/pandas/groupby.py#L222)